Implement cursor for Stories.Iterate()

devoid opened the pull request:
- Pivotal API uses offset,limit and returned headers and
  URL query parameters to paginate the result set. The
  cursor.next() function handles setting these values
  correctly to interate through each paginated endpoint.

- Use iterator for Stories.Iterate() to read all stories.

tchap commented on v5/pivotal/util.go:
```diff
@@ -0,0 +1,90 @@
+// And sets the appropriate URI variables in the next request.
+type cursor struct {
+	client *Client
+	req_fn func() (req *http.Request)
```
@devoid I don't really like this naming convention, please use CamelCase, like `reqFunc` or something...

tchap commented on v5/pivotal/stories.go:
```diff
@@ -125,6 +127,40 @@ func (service *StoryService) List(projectId int, filter string) ([]*Story, *http
+	if len(c.buff) == 0 {
+		_, err = c.next(&c.buff)
+		if err != nil {
+			return
```
@devoid I prefer to always explicitly return values, i.e. always use `return err` instead of just `err`, so please fix that in the code, it's used like that on multiple places.

devoid commented:
@tchap New patch that addresses your comments. Let me know if you would prefer a rebase, e.g. squashing the fix, or not. Also, I've got a few bigger changes in the pipeline at https://github.com/devoid/go-pivotaltracker/tree/master. These change some of the calling conventions, though so I am more hesitant to start a pull request on them.

tchap commented:
@devoid Looks good, squashing and rebasing would be nice :-)

tchap commented:
@devoid Bring it on ;-)

devoid commented:
Ok Rebased.

tchap commented on v5/pivotal/stories.go:
```diff
@@ -125,6 +127,40 @@ func (service *StoryService) List(projectId int, filter string) ([]*Story, *http
+}
+
+func (service *StoryService) Iterate(projectId int, filter string) (c *StoryCursor, err error) {
+	req_fn := func() (req *http.Request) {
```
@devoid `reqFn` please ;-)

devoid replied:
Sorry, missed that. Done.
